Lesson17/Ayse/contactBook.js

```diff
@@ -0,0 +1,331 @@
+*/
+console.log('STEP 2: Display All Contacts');
+console.log('Method-1');
+
+function displayAllContacts(contactList){
+
+  for(let contact of contactList){
+    console.log(`Name: ${contact.name}, Phone: ${contact.phone}, Email: ${contact.email}`);
+  }
+}
+displayAllContacts(contacts);
+
+console.log('Method-2');
+
+function displayAllContactsUppercase(contactList, uppercaseMode){
+  for (const contact of contactList){
+    if(uppercaseMode){
+      console.log(`NAME: ${contact.name}, PHONE: ${contact.phone}, EMAIL: ${contact.email}`,);
+      continue;
+    }else{
+      console.log(`name: ${contact.name}, phone: ${contact.phone}, email: ${contact.email}`,);
+    }
+  }
+}
+displayAllContactsUppercase(contacts, true);
+displayAllContactsUppercase(contacts, false);
+
+/*
+-----------------------------------------------------------
+  STEP 3: Add a New Contact
+-----------------------------------------------------------
+Function: addContact(name, phone, email)
+- Creates a new contact object and pushes it into 'contacts'.
+- Checks if a contact with the same name already
+  exists before adding. If found, logs a warning and returns.
+- Logs "Contact added successfully." if everything is good.
+*/
+console.log('STEP 3: Add a New Contact');
+
+
+let contacts3 = [
+  { name: 'Alice', phone: '123-456-7890', email: 'alice@gmail.com' },
+  { name: 'Anton', phone: '123-456-7899', email: 'anton@gmail.com' },
+  { name: 'Bob', phone: '222-222-2222', email: 'bob@gmail.com' }
+];
+
+function addContact(name, phone, email, contactList) {
+  for (let i = 0; i < contactList.length; i++) {
+    if (name === contactList[i].name) {
+      console.warn('The contact already exists');
+      return;
+    }
+  }
+  const contactListLengthPrev = contactList.length;
+  contactList.push({ name, phone, email });
+
+  const contactListLengthNew = contactList.length;
+
+  if (contactListLengthPrev < contactListLengthNew) {
+    console.log('Contact added successfully');
+  } else {
+    console.log('An error occurred when adding a contact');
+  }
+}
+
+  addContact("Charlie", "333-333-3333", "charlie@gmail.com", contacts3);
+  addContact("Alice", "111-111-1111", "alice_new@gmail.com", contacts3);
+  console.log(contacts3);
+
+
+/*
+-----------------------------------------------------------
+  STEP 4: View a Contact by Name
+-----------------------------------------------------------
+Function: viewContact(name)
+- Loops over 'contacts' to find one matching 'name'.
+- Logs the contact info if found.
+- Otherwise, logs: "No contact found with the name: <name>"
+*/
+
+console.log('STEP 4: View a Contact by Name');
+
+let contacts4 = [
+  { name: 'Alice', phone: '123-456-7890', email: 'alice@gmail.com' },
+  { name: 'Anton', phone: '123-456-7899', email: 'anton@gmail.com' },
+  { name: 'Bob', phone: '222-222-2222', email: 'bob@gmail.com' }
+];
+
+function viewContact(name, contactList) {
+  let isContactFound = false;
+  for (const contact of contactList) {
```

Comment:
Many functions in this application need to find a contact in the list. For that, we created a findContact function in the lesson that checks if the contact is in the list or not. Use it here instead of the loop.

I also recommend that you modify findContact a bit so it returns not true and false but contact and null. This way, you'll be able to use the return value in this function directly to print contact information.